Branch-SDK/src/io/branch/referral/ShareLinkManager.java

```diff
@@ -216,7 +218,7 @@ private void createShareDialog(List<SharingHelper.SHARE_WITH> preferredOptions)
             public void onItemClick(AdapterView<?> adapterView, View view, int pos, long l) {
                 if (view.getTag() instanceof MoreShareItem) {
                     appList_ = cleanedMatchingAppsFinal;
-                    adapter.notifyDataSetChanged();
+                    shareOptionListView.setAdapter(adapter = new ChooserArrayAdapter());
```

Comment:
@ibogolyubskiy Do you think this is needed. Each time creating the adapter again is costly

Reply:
I suggest to set new list adapter instead adapter.notifyDataSetChanged() to avoid residual animation after click on "Show more" item. 
https://www.dropbox.com/s/em2yc5mfhj031dd/residual_animation.mp4?dl=0
As you can see in video, after click on "Show more" dataset for adapter changed, but previous ripple animation for selection on "Show more" not completely finished and after changing dataset animation finishes already on "Android Beam" item.